core/include/msg.h

```diff
@@ -11,14 +11,120 @@
  * @ingroup     core
  * @brief       Messaging API for inter process communication
  *
- * There are two ways to use the IPC Messaging system of RIOT. The default is
- * synchronous messaging. In this manner, messages are either dropped when the
- * receiver is not waiting and the message was sent non-blocking, or will be
- * delivered immediately when the receiver calls msg_receive(msg_t* m). To use
- * asynchronous messaging any thread can create its own queue by calling
- * msg_init_queue(msg_t* array, int num). Messages sent to a thread with a non
- * full message queue are never dropped * and the sending never blocks. Threads
- * with a full message queue behaves like in synchronous mode.
+ * Messages
+ * ========
+ * IPC messages consist of a sender PID, a type, and some content. The sender
+ * PID will be set by the IPC internally and is not required to be set by the
+ * user. The type helps the receiver to multiplex different message types and
+ * should be set to a system-wide unique value. The content can either be
+ * provided as a 32-bit integer or a pointer.
+ *
+ * Synchronous IPC
```

Comment:
I've read it as statement that message queues are required for asynchronous IPC, but optional for what I understood as synchronous. Hopefully, my update will lift this confusion for future (and current developers).